test/unit/mem/buf/src/ConstBufTest.cpp

```diff
@@ -0,0 +1,119 @@
+    }
+#endif
+#if defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevHipRt dev, TExtent extent) -> alpaka::ConstBufHipRt<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+#if defined(ALPAKA_ACC_SYCL_ENABLED) and defined(ALPAKA_SYCL_ONEAPI_CPU)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevCpuSycl dev, TExtent extent) -> alpaka::ConstBufCpuSycl<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+#if defined(ALPAKA_ACC_SYCL_ENABLED) and defined(ALPAKA_SYCL_ONEAPI_GPU)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevGpuSyclIntel dev, TExtent extent) -> alpaka::ConstBufGpuSyclIntel<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+#if defined(ALPAKA_ACC_SYCL_ENABLED) && defined(ALPAKA_SYCL_ONEAPI_FPGA)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevFpgaSyclIntel dev, TExtent extent) -> alpaka::ConstBufFpgaSyclIntel<TElem, TDim, TIdx>
+    {
+        return alpaka::makeConstBuf(alpaka::allocBuf<TElem, TIdx>(dev, extent));
+    }
+#endif
+
+    template<typename TBuf>
+    concept onlyConstNativePtr = requires(TBuf t) {
+        { alpaka::getPtrNative(t) } -> std::same_as<alpaka::Elem<TBuf> const*>;
+    };
+
+} // namespace buftest
+
+template<typename TAcc>
+static auto testConstBuffer(alpaka::Vec<alpaka::Dim<TAcc>, alpaka::Idx<TAcc>> const& extent) -> void
+{
+    using Dev = alpaka::Dev<TAcc>;
+    using Queue = alpaka::test::DefaultQueue<Dev>;
+
+    using Elem = float;
+    using Dim = alpaka::Dim<TAcc>;
+    using Idx = alpaka::Idx<TAcc>;
+
+    auto const platformAcc = alpaka::Platform<TAcc>{};
+    auto const dev = alpaka::getDevByIdx(platformAcc, 0);
+    Queue queue(dev);
+
+    // alpaka::malloc
+    auto buf = alpaka::allocBuf<Elem, Idx>(dev, extent);
+    auto const c_buf = buftest::allocConstBuf<Dim, Elem, Idx>(dev, extent);
+
+    using TBuf = decltype(buf);
+    using TCBuf = decltype(c_buf);
+
+    auto const offset = alpaka::Vec<Dim, Idx>::zeros();
+    alpaka::test::testViewImmutable<Elem>(c_buf, dev, extent, offset);
+
+    // check that Constant buffer can't be converted to non-const buffer
+    STATIC_REQUIRE_FALSE(std::convertible_to<TCBuf, TBuf>);
+    STATIC_REQUIRE(std::convertible_to<TCBuf, TCBuf>);
+    STATIC_REQUIRE(std::convertible_to<TBuf, TBuf>);
+    STATIC_REQUIRE(std::convertible_to<TBuf, TCBuf>);
+
+    // check that Constant buffer cannot give a non const native ptr
+    // STATIC_REQUIRE(buftest::onlyConstNativePtr<TCBuf>);
+    // getPtrNative(c_buf)* = 0.f;  // <- this does not compile, as desired
```

Comment:
```suggestion
    // *getPtrNative(c_buf) = 0.f;  // <- this does not compile, as desired
```

Typo and syntax error.